golang-sample/cloudbuild.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  # Initialize the module
+  - name: golang
+    args: ['go', 'mod', 'init', 'github.com/GoogleCloudPlatform/cloud-build-samples/golang']
+
+  # Clean up dependencies
+  - name: golang
+    args: ['go', 'mod', 'tidy']
+
+  # Build application
+  - name: golang
+    args: ['go', 'build', '.']
+
+  # Run tests and save to file
+  - name: golang
+    entrypoint: /bin/bash
+    args: 
+      - -c
+      - |
+        go test . -v -json > ${SHORT_SHA}_test_log.json
+
+  # Docker Build
+  - name: 'gcr.io/cloud-builders/docker'
+    args: ['build', '-t', 
+           'us-central1-docker.pkg.dev/$PROJECT_ID/$_REPO_NAME/myimage:$SHORT_SHA', '.']
+
+  # Docker push to Google Artifact Registry
+  - name: 'gcr.io/cloud-builders/docker'
+    args: ['push',  'us-central1-docker.pkg.dev/$PROJECT_ID/$_REPO_NAME/myimage:$SHORT_SHA']
+
+  # Deploy to Cloud Run
+  - name: google/cloud-sdk
+    args: ['gcloud', 'run', 'deploy', 'helloworld', 
+           '--image=us-central1-docker.pkg.dev/$PROJECT_ID/$_REPO_NAME/myimage:$SHORT_SHA', 
+           '--region', 'us-central1', '--platform', 'managed', 
+           '--allow-unauthenticated']
```

Comment:
I have a strong preference against `--allow-unauthenticated` being used in a CD pipeline unless.

Reply:
I'm going to leave this since I want the user to end with something they can click on and see.  Maybe we'll add something in the doc about using approvals or something like that.

Reply:
One of the reasons: Should Cloud Build SA even have that IAM access?

Reply:
@nityaravi - opinion? How do you want the doc to end?  The benefit of having `--allow-unauthenticated` is they can go look at the URL at the end and see the hello world.  But this isn't a tutorial page so I can be convinced that it's not necessary that they have a working public app at the end.

Reply:
We discussed and decided to remove from this sample and to investigate it in other samples.